metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -2031,4 +2044,25 @@ private DataPlatformInfo getDataPlatformInfo(Urn urn) {
     }
     return null;
   }
+
+  private void checkConditionalUpdate(EntityAspect oldAspect, ExtraIngestParams extraIngestParams) {
+    Long updateIfCreatedOn = extraIngestParams != null && oldAspect != null
+            ? extraIngestParams.getCreatedOn(oldAspect.getUrn(), oldAspect.getAspect()) : null;
+    if (_aspectDao.supportTransactions() && updateIfCreatedOn != null) {
+      long updateIfCreatedOnValue = updateIfCreatedOn.longValue();
+      if (oldAspect != null && updateIfCreatedOnValue == 0) {
+        throw new PreconditionFailedException("Conditional Update value must be greater then 0 for update operation,"
+                + "expected value is " + oldAspect.getCreatedOn().getTime());
+      }
```

Comment:
Switching `updateIfCreatedOnValue` to be first as this is our expected state. `oldAspect` is the actual state. I think it flows more easily to think "we expect no existing value, but we found an aspect"
```suggestion
      if (updateIfCreatedOnValue == 0 && oldAspect != null) {
        throw new PreconditionFailedException("Failed to insert " + urn + "+" + aspectName
                + ". Expected no existing aspect, but found one created at " + oldAspect.getCreatedOn().getTime()
                + " by " + oldAspect.getCreatedBy());
      }
```